extensions/cornerstone-3d/src/utils/dicomRTAnnotationExport/RTStructureSet/RTSSReport.js

```diff
@@ -0,0 +1,259 @@
+   * Convert Cornerstone tool annotation data to RTSSReport report object
+   * containing the dcmjs dicom dataset
+   * @param annotations Array of Cornerstone tool annotation data
+   * @param metadataProvider Metadata provider
+   * @param options report generation options
+   * @returns Report object containing the dataset
+   */
+  static generateReport(annotations, metadataProvider, options) {
+    let dataset = initializeDataset(annotations, metadataProvider);
+
+    annotations.forEach((annotation, index) => {
+      const ContourSequence = AnnotationToPointData.convert(
+        annotation,
+        index,
+        metadataProvider,
+        options
+      );
+
+      dataset.StructureSetROISequence.push(
+        getStructureSetModule(annotation, index, metadataProvider)
+      );
+
+      dataset.ROIContourSequence.push(ContourSequence);
+      dataset.RTROIObservationsSequence.push(
+        getRTROIObservationsSequence(annotation, index, metadataProvider)
+      );
+
+      // ReferencedSeriesSequence
+      // Todo: handle more than one series
+      dataset.ReferencedSeriesSequence = getReferencedSeriesSequence(
+        annotation,
+        index,
+        metadataProvider
+      );
+
+      // ReferencedFrameOfReferenceSequence
+      dataset.ReferencedFrameOfReferenceSequence = getReferencedFrameOfReferenceSequence(
+        annotation,
+        metadataProvider,
+        dataset
+      );
+    });
+
+    const fileMetaInformationVersionArray = new Uint8Array(2);
+    fileMetaInformationVersionArray[1] = 1;
+
+    const _meta = {
+      FileMetaInformationVersion: {
+        Value: [fileMetaInformationVersionArray.buffer],
+        vr: 'OB',
+      },
+      TransferSyntaxUID: {
+        Value: ['1.2.840.10008.1.2.1'],
+        vr: 'UI',
+      },
+      ImplementationClassUID: {
+        Value: [DicomMetaDictionary.uid()], // TODO: could be git hash or other valid id
+        vr: 'UI',
+      },
+      ImplementationVersionName: {
+        Value: ['dcmjs'],
+        vr: 'SH',
+      },
+    };
+
+    dataset._meta = _meta;
+
+    return dataset;
+  }
+
+  /**
+   * Generate Cornerstone tool state from dataset
+   * @param {object} dataset dataset
+   * @param {object} hooks
+   * @param {function} hooks.getToolClass Function to map dataset to a tool class
+   * @returns
+   */
+  static generateToolState(dataset, hooks = {}) {
+    // Todo
+    console.warn('RTSSReport.generateToolState not implemented');
+  }
+}
+
+function initializeDataset(annotations, metadataProvider) {
+  const rtSOPInstanceUID = DicomMetaDictionary.uid();
+
+  // get the first annotation data
+  const {
+    referencedImageId: imageId,
+    FrameOfReferenceUID,
+  } = annotations[0].metadata;
+
+  const { studyInstanceUID } = metadataProvider.get(
+    'generalSeriesModule',
+    imageId
+  );
+
+  const patientModule = getPatientModule(imageId, metadataProvider);
+  const rtSeriesModule = getRTSeriesModule(imageId, metadataProvider);
+
+  return {
```

Comment:
Doesn't have to be right now, but this seems like it should all go down to dcmjs at some point in the future as a DICOM RTSTRUCT derivation.